### ISSUE TITLE
ERR_OSSL_DSO_COULD_NOT_LOAD_THE_SHARED_LIBRARY

### ISSUE THREAD

**Issue by sag333ar** [3speaknetwork/video-encoder#9]
```
2|video    | [Nest] 44640  - 22/06/2022, 6:20:36 am     LOG [RouterExplorer] Mapped {/api/v0/gateway/pushJob, POST} route +0ms
2|video    | [Nest] 44640  - 22/06/2022, 6:20:36 am     LOG [RouterExplorer] Mapped {/api/v0/gateway/sync, GET} route +0ms
2|video    | [Nest] 44640  - 22/06/2022, 6:20:36 am     LOG [RouterExplorer] Mapped {/api/v0/gateway/jobstatus/:job_id, GET} route +0ms
2|video    | [Nest] 44640  - 22/06/2022, 6:20:36 am     LOG [NestApplication] Nest application successfully started +1ms
2|video    | unhandledRejection Error: error:25066067:DSO support routines:dlfcn_load:could not load the shared library
2|video    |     at Sign.sign (node:internal/crypto/sig:131:29)
2|video    |     at Object.exports.hashAndSign (/home/sagar/projects/video-encoder/node_modules/libp2p-crypto/src/keys/rsa.js:65:15)
2|video    |     at RsaPrivateKey.sign (/home/sagar/projects/video-encoder/node_modules/libp2p-crypto/src/keys/rsa-class.js:64:19)
2|video    |     at Function.Envelope.seal (/home/sagar/projects/video-encoder/node_modules/libp2p/src/record/envelope/index.js:154:42)
2|video    |     at updateSelfPeerRecord (/home/sagar/projects/video-encoder/node_modules/libp2p/src/record/utils.js:21:35)
2|video    |     at EventEmitter.<anonymous> (/home/sagar/projects/video-encoder/node_modules/libp2p/src/transport-manager.js:181:40)
2|video    |     at EventEmitter.emit (node:events:527:28)
2|video    |     at Server.<anonymous> (/home/sagar/projects/video-encoder/node_modules/libp2p-tcp/src/listener.js:136:37)
2|video    |     at Server.emit (node:events:539:35)
2|video    |     at emitListeningNT (node:net:1406:10) {
2|video    |   opensslErrorStack: [
2|video    |     'error:0E076071:configuration file routines:module_run:unknown module name',
2|video    |     'error:0E07506E:configuration file routines:module_load_dso:error loading dso',
2|video    |     'error:25070067:DSO support routines:DSO_load:could not load the shared library'
2|video    |   ],
2|video    |   library: 'DSO support routines',
2|video    |   function: 'dlfcn_load',
2|video    |   reason: 'could not load the shared library',
2|video    |   code: 'ERR_OSSL_DSO_COULD_NOT_LOAD_THE_SHARED_LIBRARY'
2|video    | }
2|video    | https://encoder-gateway.infra.3speak.tv
2|video    | jobInfo {"errors":[{"message":"Cannot read property 'byte_rate' of undefined","locations":[{"line":3,"column":9}],"path":["queueJob"]}],"data":{"queueJob":null}} { node_id: 'did:key:z6Mkh9MYYAPmrUogCmwxxaGxLbwcxWtbWKyATk1TAi6qkicW' }
2|video    | unhandledRejection TypeError: Cannot read properties of null (reading 'job')
2|video    |     at GatewayClient.getNewJobs (/home/sagar/projects/video-encoder/dist/modules/core/gatewayClient.service.js:131:32)
2|video    |     at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by vaultec81** [comment]
First glance appears to be directly related to libp2p. I plan on either making it optional or remove it entirely for regular nodes. For the most part we only need HTTP connections between client nodes and gateway / cluster nodes. We could also shift the direct P2P functionality over to using IPFS pubsub as it's more efficient and works well in other areas of the project.

**Comment by vaultec81** [comment]
Fixed/mitigated in f72f8256202588ba96914fca4c4820b23fe569f0
